subsys/bluetooth/host/l2cap.c

```diff
@@ -1340,6 +1341,10 @@ static void l2cap_chan_update_credits(struct bt_l2cap_le_chan *chan,
 
 	/* Restore credits */
 	credits = chan->rx.init_credits - k_sem_count_get(&chan->rx.credits);
```

Comment:
If the channel sets init_credits to 0, then credits could become negative, but credits is unsigned.
Or how did you intend the channel to use bt_l2cap_chan_set_credits()?

Reply:
Yep, the it should never reach this line since the check above it should return iirc.